src/bin/nydusd/daemon.rs

```diff
@@ -429,35 +454,28 @@ fn fs_backend_factory(cmd: &FsBackendMountCmd) -> DaemonResult<BackFileSystem> {
 // - `Init` means nydusd is just started and potentially configured well but not
 //    yet negotiate with kernel the capabilities of both sides. It even does not try
 //    to set up fuse session by mounting `/fuse/dev`(in case of `fusedev` backend).
+// - `Ready` means nydusd is ready for start or die. Fuse session is created.
 // - `Running` means nydusd has successfully prepared all the stuff needed to work as a
 //   user-space fuse filesystem, however, the essential capabilities negotiation might not be
 //   done yet. It relies on `fuse-rs` to tell if capability negotiation is done.
-// - `Upgrading` state means the nydus daemon is being live-upgraded. There's no need
-//   to do kernel mount again to set up a session but try to reuse a fuse fd from somewhere else.
-//   In this state, we try to push `Successful` event to state machine to trigger state transition.
-// - `Interrupt` state means nydusd has shutdown fuse server, which means no more message will
-//    be read from kernel and handled and no pending and in-flight fuse message exists. But the
-//    nydusd daemon should be alive and wait for coming events.
 // - `Die` state means the whole nydusd process is going to die.
 state_machine! {
     derive(Debug, Clone)
     pub DaemonStateMachine(Init)
 
-    // FIXME: It's possible that failover does not succeed or resource is not capable to
-    // be passed. To handle event `Stop` when being `Init`.
     Init => {
-        Mount => Running [StartService],
-        Takeover => Upgrading [Restore],
-        Exit => Die[StopStateMachine],
+        Mount => Ready,
```

Comment:
> Seems we no longer need `Mount` event since nothing is done when migrating state to Ready

In fact, we shall do `Mount` action. However, it's a bit risky now. Because the Mount content are different in different cases, such as upgrade startup, normal startup, endpoint startup. In total, You're right. There will be abstraction for `Mount` action in future.